yesod-bin/Devel.hs

```diff
@@ -222,6 +225,33 @@ checkDevelFile =
             then return x
             else loop xs
 
+stackSuccessString :: ByteString
+stackSuccessString = "ExitSuccess"
+
+stackFailureString :: ByteString
+stackFailureString = "ExitFailure"
+
+data BuildOutput = Started
+                   deriving (Show, Eq, Ord)
+
+makeEmptyMVar :: MVar a -> IO ()
+makeEmptyMVar mvar = do
+  isEmpty <- isEmptyMVar mvar
+  case isEmpty of
+    True -> return ()
+    False -> takeMVar mvar >> return ()
+
+updateAppPort :: ByteString -> MVar (BuildOutput) -> TVar Int -> IO ()
+updateAppPort bs mvar appPortVar = do
+  isEmpty <- isEmptyMVar mvar
```

Comment:
It seems like the logic here would be easier to follow as a TVar Bool instead of MVar Build output.